configure.sh

```diff
@@ -278,7 +278,7 @@ gen_file_lists() {
 
 	if [ "$_gen_file_lists_use" -ne 0 ]; then
 
-		_gen_file_lists_replacement=$(ls $_gen_file_lists_filedir/*.c | tr '\n' ' ')
+		_gen_file_lists_replacement=$(cd "$_gen_file_lists_filedir" && find . ! -name . -prune -name "*.c" | cut -d/ -f2 | sed "s@^@$_gen_file_lists_filedir/@g" | tr '\n' ' ')
 		_gen_file_lists_contents=$(replace "$_gen_file_lists_contents" "$_gen_file_lists_needle_src" "$_gen_file_lists_replacement")
```

Comment:
What's so "dangerous" about the `ls` call?

Reply:
As talked about on IRC, this particular use of globbing is not dangerous since the files with the `.c` extension in the directories are well known, but the `ls` is redundant. I'll take care of that in my own changes.